app/assets/javascripts/admin/products/services/unit_prices.js.coffee

```diff
@@ -0,0 +1,25 @@
+angular.module("admin.products").factory "UnitPrices", (VariantUnitManager) ->
+  class UnitPrices
+    @price: (price, scale, unit_type, unit_value, variant_unit_name = '') ->
+      price / @denominator(scale, unit_type, unit_value, variant_unit_name)
+
+    @denominator: (scale, unit_type, unit_value, variant_unit_name = '') ->
+      unit = @unit(scale, unit_type, variant_unit_name)
+      if unit == "lb"
+        unit_value / 453.6
+      else if unit == "kg"
+        unit_value / 1000
+      else
+        unit_value
+
+    @unit: (scale, unit_type, variant_unit_name = '') ->
+      if variant_unit_name.length > 0
+        variant_unit_name
+      else if unit_type == "items"
+        "item"
+      else if VariantUnitManager.systemOfMeasurement(scale, unit_type) == "imperial"
+        "lb"
+      else if unit_type == "weight"
+        "kg"
+      else if unit_type == "volume"
+        "L"
```

Comment:
These changes look fine - you know better than I do what will be useful. My original intent, however, was to have the `denominator` function return the number that we need to divide the price by, and the `unit` function return the string that we present to the user. For example, a 500g item that sells for $5 would have its `denominator` be 0.5, and its price be $10, and its unit be `kg`. That's how the backend is doing it; it would be good if they're consistent. 

Maybe you're heading in that direction anyway and I'm missing something though. In that case there's no need to pass `variant_unit_name` to `denominator`.

Reply:
I see what you mean and it makes sense to me. I've updated the file, and I think it's better and consistent now.